Release notes for Foreman 3.1(GA) (#1932)
theforeman/theforeman.org: _includes/manuals/3.1/1.2_release_notes.md

* ##### New Build button and card for BMC Power Management
New modals for a build button and BMC Power Management is available, which improves exprience of new Host Details page.

#### Descriptions for all of the provisioning templates
Over a hundred new descriptions have been added to provisioning templates, snippets, and scripts to assist users understand particular usage and use them in an effective manner.

#### New purge:puppet rake task

> If you don't intent to use the Puppet ENC functionality then this rake task can be used to purge Environments, Puppetclasses, Smartclass parameters and Config groups.

Reviewer: ```suggestion
If you don't intend to use the Puppet ENC functionality then this rake task can be used to purge Environments, Puppetclasses, Smartclass parameters and Config groups.
```

#### Provisioning snippets support Puppet 7
The provisioning snippets have been updated to setup Puppet 7 repositories, as well as configure and install packages from those repositories.

> #### Performance improvements for index pages and Host Config Status

Author: @ekohl or @ezr-ondrej request to review this point. I think maybe I have added more details than required?

Reviewer: For a headline feature it does have a lot of details. I'd try to keep it shorter and leave out some implementation details. Focus on the results for the end user.

### Deprecations

#### The `:unattended => false` Setting has been deprecated

> The host registration way can be used to register hosts on Foreman for the sole purpose of inventory. Refer [the RFC](https://community.theforeman.org/t/rfc-remove-unattended-setting/10035) for more information.

Reviewer: ```suggestion
The host registration feature can be used to register hosts on Foreman for the sole purpose of an inventory. Refer to [the removal RFC](https://community.theforeman.org/t/rfc-remove-unattended-setting/10035) for more information.
```

### Deprecations

> #### The `:unattended => false` Setting has been deprecated

Author: and also this one, as it has been cherry picked in 3.1

#### Foreman - Authentication
* Appropriate error message to be shown when creating authsource with same name as existing authsource. \- [#33248](https://projects.theforeman.org/issues/33248)

> #### Foreman - BMCc

Reviewer: This looks accidental.

#### Foreman - Compute resources - oVirt
* Foreman error on retrieving networks/vnics from oVirt if description contains non ascii characters \- [#32908](https://projects.theforeman.org/issues/32908)

#### Foreman - DHCP

> * Report nicely when subnet has invalid address \- [#33929](https://projects.theforeman.org/issues/33929)

Reviewer 2: I know this is the title, so unsure if you want to change or not, but 
```suggestion
* Provide a meaningful report when a subnet has invalid address \- [#33929](https://projects.theforeman.org/issues/33929)
```
I think this better explains. It is not a blocker in my view.

> * Provide a meaningful report when a subnet has invalid address \- [#33929](https://projects.theforeman.org/issues/33929)

Reviewer: I also changed the Redmine issue:
```suggestion
* Provide a meaningful report when a subnet has an invalid address \- [#33929](https://projects.theforeman.org/issues/33929)
```

Author: Updated

Reviewer: It doesn't look updated.

Author: done

* Remove hard-coded link to documentation \- [#33215](https://projects.theforeman.org/issues/33215)
* Move katello_ca_consumer to registration template \- [#32678](https://projects.theforeman.org/issues/32678)

#### Foreman - Internationalization
* Localized string broken into two separate lines, at &lt;Administer - Locations page&gt;, at &lt;Administer - Organizations page&gt; and at &lt;Infrastructure - Domains&gt;. \- [#31338](https://projects.theforeman.org/issues/31338)

#### Foreman - Inventory
* Handle different Windows installation types \- [#33769](https://projects.theforeman.org/issues/33769)
* Distinguish between "regular" hosts and hosts on which Foreman and/or smart proxy runs - models and fact processing \- [#32328](https://projects.theforeman.org/issues/32328)

#### Foreman - JavaScript stack

> * webpack can't find loaders for plugins in production \- [#34086](https://projects.theforeman.org/issues/34086)

Reviewer 2: ```suggestion
* Webpack can't find loaders for plugins in production \- [#34086](https://projects.theforeman.org/issues/34086)
```

* TFTP Unable to download Flatcar OS \- [#33469](https://projects.theforeman.org/issues/33469)

#### Foreman - Templates

> * come up with slightly better name for puppet7-repo parameter \- [#34088](https://projects.theforeman.org/issues/34088)

Reviewer 2: ```suggestion
* Come up with slightly better name for puppet7-repo parameter \- [#34088](https://projects.theforeman.org/issues/34088)
```

> * Come up with slightly better name for puppet7-repo parameter \- [#34088](https://projects.theforeman.org/issues/34088)

Reviewer: I've renamed this in Redmine. I also think this is a breaking change since it lacks a DB migration. That means we should at least add an upgrade warning that we'll break provisioning setups. Noted in https://github.com/theforeman/foreman/pull/8970#issuecomment-990044976 too.

Reviewer 3: No need, this was new parameter in 3.1

Reviewer: Thanks for checking. Sounds like it could only affect users of RCs and we never give them guarantees so that's acceptable.
